synapse/handlers/sync.py

```diff
@@ -794,7 +794,7 @@ def compute_state_delta(self, room_id, batch, sync_config, since_token, now_toke
             else:
                 state_ids = {}
                 if lazy_load_members:
-                    if types:
+                    if types and batch.events:
```

Comment:
why would `types` be empty here if `lazy_load_members` was true, now that #3916 is a thing?

Reply:
it wouldn't.